src/media/js/elements/header.js

```diff
@@ -0,0 +1,156 @@
+    var CHILD_VISIBLE = 'mkt-header-child--visible';
+    // Data attribute for whether the header child contains an input.
+    var CHILD_INPUT = 'data-header-child--input';
+
+    var MktHeaderElement = document.registerElement('mkt-header', {
+        prototype: Object.create(HTMLElement.prototype, {
+            createdCallback: {
+                value: function() {
+                    var root = this;
+
+                    // Move <mkt-header-child>ren out since we need to z-index
+                    // them below <mkt-header>.
+                    var headerChildren = root.headerChildren();
+                    for (var i = 0; i < headerChildren.length; i++) {
+                        root.parentNode.insertBefore(headerChildren[i],
+                                                     root.nextSibling);
+                    }
+                }
+            },
+            headerChildren: {
+                value: function() {
+                    return document.querySelectorAll('mkt-header-child');
+                }
+            },
+            hideHeaderChildren: {
+                value: function() {
+                    var headerChildren = this.headerChildren();
+                    for (var i = 0; i < headerChildren.length; i++) {
+                        headerChildren[i].toggle(false);
+                    }
+                }
+            }
+        })
+    });
+
+    var MktHeaderChildElement = document.registerElement('mkt-header-child', {
+        prototype: Object.create(HTMLElement.prototype, {
+            createdCallback: {
+                value: function() {
+                    var root = this;
+
+                    if (root.hasAttribute(CHILD_INPUT)) {
+                        // If main element is input, then hide child on blur.
+                        root.querySelector('input').onblur = function() {
+                            setTimeout(function() {
+                                // setTimeout to queue behind toggle button.
+                                root.toggle(false);
+                            }, 50);
+                        };
+                    }
+                }
+            },
+            toggle: {
+                value: function(bool) {
+                    // Toggle visibility.
+                    var root = this;
+                    if (bool !== undefined) {
+                        if (bool) {
+                            root.classList.add(CHILD_VISIBLE);
+                        } else {
+                            root.classList.remove(CHILD_VISIBLE);
+                        }
+                    } else {
+                        root.classList.toggle(CHILD_VISIBLE);
+                    }
+
+                    // Keep state on the toggle button. The child is source of
+                    // truth so keep this logic in the child.
+                    var toggleButton = document.querySelector(
+                        'mkt-header-child-toggle[for="' + root.id + '"]');
```

Comment:
I think we should be using more events. If they're a pain to use we should setup some helpers so it's easier.

I don't think events means it's optional, it just means that any number of elements can react to this change. To me it makes sense for the button to have logic like "when the search bar opens, be in the active state, when it hides, leave the active state".